web-services/query/src/main/java/datawave/webservice/query/metric/QueryMetricsWriter.java

```diff
@@ -64,285 +52,152 @@
     private Queue dest;
     
     @Inject
-    private AccumuloConnectionFactory connectionFactory;
-    
-    @Inject
-    private QueryMetricHandler<? extends BaseQueryMetric> queryMetricHandler;
-    
-    @Inject
-    @SpringBean(name = "QueryMetricsWriterConfiguration", refreshable = true)
-    private QueryMetricsWriterConfiguration config;
+    private RemoteQueryMetricService remoteQueryMetricService;
     
-    private UdpClient timelyClient = null;
-    private Map<String,Long> lastPageMetricMap;
-    
-    // queryId to lastPage Map
-    private Map<String,Long> lastPageMap;
-    private List<QueryMetricHolder> metricQueue;
-    private DecimalFormat df = new DecimalFormat("0.00");
+    private List<FailureRecord> failedMetricList = new ArrayList<>();
     
     private static volatile AtomicBoolean receivingMetrics = new AtomicBoolean(false);
     
-    private UdpClient createUdpClient() {
-        if (config != null && StringUtils.isNotBlank(config.getTimelyHost())) {
-            return new UdpClient(config.getTimelyHost(), config.getTimelyPort());
-        } else {
-            return null;
+    private List<QueryMetricHolder> getMetricsFromQueue() {
+        List metricHolderList = new ArrayList<>();
+        long start = System.currentTimeMillis();
+        try (JMSConsumer consumer = jmsContext.createConsumer(dest)) {
+            Message message;
+            do {
+                message = consumer.receive(500);
+                if (message != null) {
+                    try {
+                        if (message instanceof ObjectMessage) {
+                            ObjectMessage objectMessage = (ObjectMessage) message;
+                            Object o = objectMessage.getObject();
+                            QueryMetricHolder queryMetricHolder = null;
+                            if (o instanceof QueryMetricHolder) {
+                                queryMetricHolder = (QueryMetricHolder) o;
+                            } else if (o instanceof QueryMetricMessage) {
+                                queryMetricHolder = ((QueryMetricMessage) o).getMetricHolder();
+                            }
+                            if (queryMetricHolder != null) {
+                                metricHolderList.add(queryMetricHolder);
+                            }
+                        }
```

Comment:
I'm wondering if it's worth logging when the message is not an ObjectMessage.

Reply:
Added logging